port/unix/omrsysinfo.c

```diff
@@ -3613,25 +3710,23 @@ omrsysinfo_cgroup_is_system_available(struct OMRPortLibrary *portLibrary)
 	int32_t rc = OMRPORT_ERROR_SYSINFO_CGROUP_UNSUPPORTED_PLATFORM;
 
 	if (NULL == PPG_cgroupEntryList) {
-		struct statfs buf = {0};
-
-		/* If tmpfs is mounted on /sys/fs/cgroup, then it indicates cgroup v1 system is available */
-		rc = statfs(OMR_CGROUP_V1_MOUNT_POINT, &buf);
-		if (0 != rc) {
-			rc = portLibrary->error_set_last_error(portLibrary, errno, OMRPORT_ERROR_SYSINFO_SYS_FS_CGROUP_STATFS_FAILED);
-			goto _end;
-		} else if (TMPFS_MAGIC != buf.f_type) {
-			rc = portLibrary->error_set_last_error_with_message_format(portLibrary, OMRPORT_ERROR_SYSINFO_SYS_FS_CGROUP_TMPFS_NOT_MOUNTED, "tmpfs is not mounted on " OMR_CGROUP_V1_MOUNT_POINT);
-			goto _end;
-		}
-
-		omrthread_monitor_enter(cgroupEntryListMonitor);
-		if (NULL == PPG_cgroupEntryList) {
-			rc = readCgroupFile(portLibrary, getpid(), &PPG_cgroupEntryList, &PPG_cgroupSubsystemsAvailable);
-		}
-		omrthread_monitor_exit(cgroupEntryListMonitor);
-		if (0 != rc) {
-			goto _end;
+		if (isCgroupV1Available(portLibrary)) {
```

Comment:
Do I read this code correctly that the cgroup detection is only possible on cgroup v1 systems?

Reply:
For cgroup v2, the file system type mounted on `/sys/fs/cgroup` is different. The function `isCgroupV1Available()` is currently only handling cgroup v1, therefore I named it so.
As and when we add support for cgroup v2, I believe we can rename the function then.